### PR TITLE
Cases Search UI and Bug Fixes

### PR DESCRIPTION
**Pull request by bpowers1215**
Add additional Case Work Items filters: outcome, description, status
Bug Fix: Remove closed timestamp filter
Bug Fix: Correct closed case outcome value
Bug Fix: Correctly save evidence to case work item outcome

### CODE REVIEW

File: gui/admin-gui/src/main/java/com/evolveum/midpoint/web/page/admin/cases/PageCaseWorkItems.java [hunk 1]
```diff
@@ -147,22 +151,54 @@ private ObjectQuery createQuery() throws SchemaException, ObjectNotFoundExceptio
         } else {
             // not authorized to see all => sees only allocated to him (not quite what is expected, but sufficient for the time being)
             query = QueryUtils.filterForAssignees(q, SecurityUtils.getPrincipalUser(),
-                    OtherPrivilegesLimitationType.F_APPROVAL_WORK_ITEMS)
-                    .and().item(CaseWorkItemType.F_CLOSE_TIMESTAMP).isNull().build();
+                    OtherPrivilegesLimitationType.F_APPROVAL_WORK_ITEMS).build();
```
Comment (mederly) [inline gui/admin-gui/src/main/java/com/evolveum/midpoint/web/page/admin/cases/PageCaseWorkItems.java]: `OtherPrivilegesLimitationType.F_CASE_MANAGEMENT_WORK_ITEMS)` should be here instead.

File: model/model-impl/src/main/java/com/evolveum/midpoint/model/impl/controller/ModelController.java [hunk 2]
```diff
@@ -2171,7 +2172,11 @@ public void completeWorkItem(@NotNull String caseOid, long workItemId, AbstractW
 			}
 			XMLGregorianCalendar now = XmlTypeConverter.createXMLGregorianCalendar(System.currentTimeMillis());
 			ObjectDelta<CaseType> delta = DeltaBuilder.deltaFor(CaseType.class, prismContext)
-					.item(CaseType.F_WORK_ITEM, workItemId, WorkItemType.F_OUTPUT).replace(output)
```
Comment (mederly) [inline model/model-impl/src/main/java/com/evolveum/midpoint/model/impl/controller/ModelController.java]: What was the reason of this change?

File: repo/repo-sql-impl/src/main/java/com/evolveum/midpoint/repo/sql/data/common/RCase.java [hunk 3]
```diff
@@ -54,6 +54,7 @@
     private RPolyString nameCopy;
 
     private String state;
+    private String description;
```
Comment (mederly) [inline repo/repo-sql-impl/src/main/java/com/evolveum/midpoint/repo/sql/data/common/RCase.java]: Do we really need to search by description? Maybe full text search would be a better solution.

File: repo/repo-sql-impl/src/main/java/com/evolveum/midpoint/repo/sql/data/common/container/RCaseWorkItem.java [hunk 4]
```diff
@@ -66,6 +66,7 @@
     private Set<RCaseWorkItemReference> assigneeRef = new HashSet<>();
     private REmbeddedReference performerRef;
     private String outcome;
+    private String comment;
```
Comment (mederly) [inline repo/repo-sql-impl/src/main/java/com/evolveum/midpoint/repo/sql/data/common/container/RCaseWorkItem.java]: Do we really need to search by comment? Maybe full text search would be a better solution.

**Comment by martin-lizner** [comment]
Please finish this PR asap so it goes to 3.9. Thanks.

**Comment by littleBigMen** [comment]
We would like to use cases on our project, +1 for 3.9 merge

**Comment by semancik** [comment]
This won't get merged into 3.9. Contributors haven't responded to review in time for inclusion in 3.9. But hopefully it can be merged in 4.0.

**Comment by bpowers1215** [comment]
Hi all, as @semancik mentioned, we missed the cut-off for the 3.9 version.  Unfortunately, we have not yet had availability to address the concerns that @mederly stated above.  Short of dismissing the concerns in the short term, Full Text search will require replacing existing panels implemented in the Cases GUI with those that support full-text-search - an undertaking we haven't yet had an opportunity to fully investigate.  We hope to get back to this soon.

**Comment by martin-lizner** [comment]
@bpowers1215 forgive my urging, but is there ETA for finishing this PR? Also do you really need to make changes to DB schema? Thanks.

**Comment by alexandrezia** [comment]
Awesome, great!
Thanks for this.

**Comment by martin-lizner** [comment]
@bpowers1215 thanks again for splitting the necessary fixes out of this PR. Hopefully, you will be able to merge GUI improvements eventually (cant wait :-). Seems like this feature got traction throughout the community and Exclamation Labs will make its name with it :-)

**Comment by semancik** [comment]
The case management UI was completely rewritten in midPoint 4.0. The code was aligned with the rest of the GUI and expanded with new use cases for CaseType (e.g. approvals). Therefore this pull request does not make much sense for 4.0 any more. We will not merge it into master. This PR may still be useful for 3.9. But in that case the base branch should be changed to support-3.9.

**Comment by bpowers1215** [comment]
Rado,
This is fantastic news, especially since we have not yet had the time to
address this. In fact, your email couldnt have come at a better time as I
was starting discussion with a colleague about coming back to this.  This
saves us from spending much effort needlessly.  We will be checking out the
new case UI as you mentioned in your email and will let you know how
testing goes.

Thanks!

Brandon Powers

**Comment by martin-lizner** [comment]
Will checkout the new GUI soon, thanks all!!!

**Pull request closed without merging**